src/common/multi_file_reader.cpp

```diff
@@ -240,15 +240,18 @@ void MultiFileReader::FinalizeBind(const MultiFileReaderOptions &file_options, c
 void MultiFileReader::CreateNameMapping(const string &file_name, const vector<LogicalType> &local_types,
                                         const vector<string> &local_names, const vector<LogicalType> &global_types,
                                         const vector<string> &global_names, const vector<column_t> &global_column_ids,
-                                        MultiFileReaderData &reader_data, const string &initial_file) {
+                                        MultiFileReaderData &reader_data, const string &initial_file,
+                                        const bool with_ordinality) {
 	D_ASSERT(global_types.size() == global_names.size());
 	D_ASSERT(local_types.size() == local_names.size());
 	// we have expected types: create a map of name -> column index
 	case_insensitive_map_t<idx_t> name_map;
 	for (idx_t col_idx = 0; col_idx < local_names.size(); col_idx++) {
 		name_map[local_names[col_idx]] = col_idx;
 	}
-	for (idx_t i = 0; i < global_column_ids.size(); i++) {
+	size_t num_column_ids;
```

Comment:
Hey @taniabogatsch, I have had an accident and was out in the hospital for a while, but will now continue my work. 

Can you maybe link me a resource where I can read up on the internals of DuckDB parallelism? I'll get it done either way, just not sure where to start right now.

Thank you for the feedback!

Reply:
Hi @niykko! I am very sorry to hear that you had an accident - I hope that your recovery is going well!

I discussed making `WITH ORDINALITY` multithreaded with a colleague. It will take significant effort and is outside of this PR's scope. Instead, let's make the `PhysicalTableInOutFunction` single-threaded if the user did specify `WITH ORDINALITY`.

To make it single-threaded, you must return `false` here (`physical_tableinout_function.hpp`) in the `WITH ORDINALITY` case. This should tell the previous operator to pass data single-threadedly.
```cpp
	bool ParallelOperator() const override {
		return true;
	}
```